ale_linters/c/gcc.vim

```diff
@@ -10,14 +10,24 @@ if !exists('g:ale_c_gcc_options')
 endif
 
 function! ale_linters#c#gcc#GetCommand(buffer) abort
+    " attempt to get args from compilation database
+    let l:args = ale#c#FindCompileArgs(a:buffer)
+
+    " if we've found compile args then just use those
+    if (!empty(l:args))
+        return 'gcc -S -x c -fsyntax-only '
+                    \	. l:args
+                    \   . ' -'
```

Comment:
Have a look through files and get the indentation to match the rest of the codebase.